src/axom/quest/Shaper.cpp

```diff
@@ -30,13 +27,133 @@ constexpr double Shaper::MINIMUM_PERCENT_ERROR;
 constexpr double Shaper::MAXIMUM_PERCENT_ERROR;
 constexpr double Shaper::DEFAULT_VERTEX_WELD_THRESHOLD;
 
-Shaper::Shaper(const klee::ShapeSet& shapeSet, sidre::MFEMSidreDataCollection* dc)
-  : m_shapeSet(shapeSet)
+#if defined(AXOM_USE_MFEM)
+Shaper::Shaper(RuntimePolicy execPolicy,
+               int allocatorId,
+               const klee::ShapeSet& shapeSet,
+               sidre::MFEMSidreDataCollection* dc)
+  : m_execPolicy(execPolicy)
+  , m_allocatorId(allocatorId != axom::INVALID_ALLOCATOR_ID
+                    ? allocatorId
+                    : axom::policyToDefaultAllocatorID(execPolicy))
+  , m_shapeSet(shapeSet)
   , m_dc(dc)
+  #if defined(AXOM_USE_CONDUIT)
+  , m_bpGrp(nullptr)
+  , m_bpTopo()
+  , m_bpNodeExt(nullptr)
+  , m_bpNodeInt()
+  #endif
 {
-#if defined(AXOM_USE_MPI) && defined(MFEM_USE_MPI)
+  #if defined(AXOM_USE_MPI) && defined(MFEM_USE_MPI)
   m_comm = m_dc->GetComm();
+  #endif
+  m_cellCount = m_dc->GetMesh()->GetNE();
+
+  setFilePath(shapeSet.getPath());
+}
+#endif
+
+Shaper::Shaper(RuntimePolicy execPolicy,
+               int allocatorId,
+               const klee::ShapeSet& shapeSet,
+               sidre::Group* bpGrp,
+               const std::string& topo)
+  : m_execPolicy(execPolicy)
+  , m_allocatorId(allocatorId != axom::INVALID_ALLOCATOR_ID
+                    ? allocatorId
+                    : axom::policyToDefaultAllocatorID(execPolicy))
+  , m_shapeSet(shapeSet)
+#if defined(AXOM_USE_CONDUIT)
+  , m_bpGrp(bpGrp)
+  , m_bpTopo(topo.empty() ? bpGrp->getGroup("topologies")->getGroupName(0) : topo)
+  , m_bpNodeExt(nullptr)
+  , m_bpNodeInt()
+#endif
+  , m_comm(MPI_COMM_WORLD)
```

Comment:
Maybe outside the scope of this PR but in the CMakeLists.txt, there are some files that are only enabled when AXOM_USE_MPI is enabled. This file is always enabled and it contains the m_comm member. Can quest build without MPI? It looked like before the communicator was mostly accessed through the MFEM data collection.

Reply:
Ah, didn't know about the MPI_Comm wrapper.